Feature window android

Sumenia opened the pull request:
Deleting android/log until we have the cross platform logger
Deleting unnecessary events until we have uniform cross platform events

Commit:
basic window android implementation

Commit:
small fix

Commit:
better handling of external libraries dependancies

Commit:
External library fix + better global android scope

Better APPEND for external library in Window CMakeList
Better name for Android class and better global scope (moving the event
queue from WindowImpl to MainAndroid made made more sense)

Commit:
Deleting useless code in order to merge

Deleting android/log until we have the cross platform logger
Deleting unnecessary events until we have uniform cross platform events

Quentin01 commented on samples/window/CMakeLists.txt:
```diff
@@ -1,22 +1,48 @@
+    endif()
+elseif(LUG_COMPILER_GCC OR LUG_COMPILER_CLANG)
+    list(APPEND CMAKE_CXX_FLAGS "-Wall -Wextra -Werror -std=c++14")
+    string(REPLACE ";" " " "CMAKE_CXX_FLAGS" "${CMAKE_CXX_FLAGS}")
```
Do that outside of the if. See https://github.com/Lugdunum3D/Lugdunum/blob/dev/samples/base/CMakeLists.txt#L30

Sumenia replied:
I took the code from base at that time, we edited it after but indeed I didn't apply the change here

Quentin01 commented on samples/window/CMakeLists.txt:
```diff
@@ -1,22 +1,48 @@
+    add_library(${EXECUTABLE_NAME}
+            SHARED
+            main.cpp
+            )
```
Bad indentation

Quentin01 commented on samples/window/CMakeLists.txt:
```diff
@@ -1,22 +1,48 @@
+            log
+            android
+            ${LUG_LIBRARIES}
+            )
```
Bad indentation

Quentin01 commented on samples/window/main.cpp:
```diff
@@ -1,5 +1,5 @@
-#include <lug/Config.hpp>
 #include <lug/Window/Window.hpp>
+#include <string>
```
Remove that include

Quentin01 commented on samples/window/main.cpp:
```diff
@@ -15,11 +15,9 @@ int main() {
             if (event.type == lug::Window::EventType::CLOSE) {
                 window->close();
             }
-
```
Why have you remove that line?

Sumenia replied:
what is the need of using a vertical space here ?

Quentin01 replied:
I don't know, but okay, this one is really useless

Quentin01 commented on samples/window/main.cpp:
```diff
@@ -15,11 +15,9 @@ int main() {
 
         // ...
     }
-    
```
Why have you remove that line?

Quentin01 commented on src/lug/Main/MainAndroid.hpp:
```diff
@@ -2,15 +2,19 @@
 namespace Main {
 
-class LugAndroidApp {
+extern std::queue<lug::Window::Event> events;
```
Why are you declaring them extern here? `MainAndroid.cpp` doesn't need that.

Sumenia replied:
I spent 1 hour trying to compile on android with the extern in lug/Window/Android/WindowImplAndroid.cpp with no success , ended up putting in here and it works

Quentin01 commented on src/lug/Window/Android/WindowImplAndroid.cpp:
```diff
@@ -0,0 +1,39 @@
+namespace Window {
+namespace priv {
+
+lug::Window::priv::WindowImpl::WindowImpl(Window*) {}
```
You don't need to specify `lug::Window::priv::` here

Quentin01 commented on src/lug/Window/Android/WindowImplAndroid.cpp:
```diff
@@ -0,0 +1,39 @@
+
+lug::Window::priv::WindowImpl::WindowImpl(Window*) {}
+
+bool lug::Window::priv::WindowImpl::create(const std::string&, Style) {
```
You don't need to specify `lug::Window::priv::` here

Quentin01 commented on src/lug/Window/Android/WindowImplAndroid.cpp:
```diff
@@ -0,0 +1,39 @@
+    return true;
+}
+
+void lug::Window::priv::WindowImpl::close() {}
```
You don't need to specify `lug::Window::priv::` here

Quentin01 commented on src/lug/Window/Android/WindowImplAndroid.cpp:
```diff
@@ -0,0 +1,39 @@
+
+void lug::Window::priv::WindowImpl::close() {}
+
+bool lug::Window::priv::WindowImpl::pollEvent(lug::Window::Event& event) {
```
You don't need to specify `lug::Window::priv::` here

Quentin01 commented on src/lug/Window/Android/WindowImplAndroid.cpp:
```diff
@@ -0,0 +1,39 @@
+                continue;
+            }
+            int32_t handled = 0;
+            AInputQueue_finishEvent(lug::Main::inputQueue, androidEvent, handled);
```
Why not `AInputQueue_finishEvent(lug::Main::inputQueue, androidEvent, 0);` ?

Quentin01 commented on src/lug/Window/Android/WindowImplAndroid.cpp:
```diff
@@ -0,0 +1,39 @@
+            if (AInputQueue_preDispatchEvent(lug::Main::inputQueue, androidEvent)) {
+                continue;
+            }
+            int32_t handled = 0;
```
Maybe one vertical space?

Quentin01 commented on src/lug/Window/Android/WindowImplAndroid.cpp:
```diff
@@ -0,0 +1,39 @@
+
+    if (lug::Main::inputQueue != nullptr) {
+        AInputEvent *androidEvent = nullptr;
+        while (AInputQueue_getEvent(lug::Main::inputQueue, &androidEvent) >= 0) {
```
Maybe one vertical space?

Quentin01 commented on src/lug/Window/Android/WindowImplAndroid.hpp:
```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include <lug/Window/Window.hpp>
+#include <lug/Main/MainAndroid.hpp>
```
Do you really need that include? Can't you declare your extern here?

Quentin01 commented on src/lug/Window/Android/WindowImplAndroid.hpp:
```diff
@@ -0,0 +1,23 @@
+    bool create(const std::string& title, Style style);
+    void close();
+    bool pollEvent(lug::Window::Event& event);
+private:
```
Remove that `private` if it isn't used

Quentin01 commented on src/lug/Window/Android/WindowImplAndroid.hpp:
```diff
@@ -0,0 +1,23 @@
+
+} // namespace priv
+} // namespace Window
+} // namespace lug
```
New line at end of file

Quentin01 commented on src/lug/Window/CMakeLists.txt:
```diff
@@ -16,16 +16,22 @@ set(INC
     list(APPEND SRC ${SRCROOT}/Win32/WindowImplWin32.cpp)
+elseif(LUG_OS_ANDROID)
+    include_directories(${CMAKE_SYSROOT}/usr/include)
+    list(APPEND EXT_LIBRARIES android)
```
Can't you append the two items in one line?

Commit:
Class name and static

better android class name
event queue as static class member rather than global

Commit:
Small fixes for android

Commit:
Small tweak

Quentin01 committed:
Fix some problems with the guidelines
